changelog/core-memory-__delete.dd

```diff
@@ -0,0 +1,29 @@
+`core.memory.__delete` has been added
+
+$(REF __delete, core, memory) allows easy migration from the deprecated `delete`.
+`__delete` behaves exactly like `delete` and will call the class dtor immediately
```

Comment:
`delete`/`__delete` apply to structs and arrays as well, so please update the documentation